libs/wire-api/src/Wire/API/Routes/Internal/Brig.hs

```diff
@@ -664,6 +669,81 @@ type TeamsAPI =
         :> ReqBody '[Servant.JSON] (Multi.TeamStatus SearchVisibilityInboundConfig)
         :> Post '[Servant.JSON] ()
     )
+    :<|> InvitationByEmail
+    :<|> InvitationCode
+    :<|> SuspendTeam
+    :<|> UnsuspendTeam
+    :<|> TeamSize
+    :<|> TeamInvitations
+
+type InvitationByEmail =
+  Named
+    "get-invitation-by-email"
+    ( "teams"
+        :> "invitations"
+        :> "by-email"
+        :> QueryParam' [Required, Strict] "email" Email
+        :> Get '[Servant.JSON] Invitation
+    )
+
+type InvitationCode =
+  Named
+    "get-invitation-code"
+    ( "teams"
+        :> "invitation-code"
+        :> QueryParam' [Required, Strict] "team" TeamId
+        :> QueryParam' [Required, Strict] "invitation_id" InvitationId
+        :> Get '[Servant.JSON] FoundInvitationCode
+    )
+
+newtype FoundInvitationCode = FoundInvitationCode User.InvitationCode
+  deriving stock (Eq, Show, Generic)
+  -- TODO: is this correct?
+  deriving newtype (S.ToSchema)
+
+instance ToJSON FoundInvitationCode where
+  toJSON (FoundInvitationCode c) = Aeson.object ["code" Aeson..= c]
```

Comment:
I also would like to ask if this is correct, especially the `S.ToSchema` part which does not adhere to the `ToJSON` instance but I don't know how / if they should be related.